src/parse_nlp.jl

```diff
@@ -11,6 +11,66 @@ function _let_code_block(ex::Expr)
     return ex.args[2]
 end
 
+function _try_parse_idx_set(arg::Expr)
```

Comment:
Are these functions used only in `parse_nlp.jl`? If they're also used elsewhere, this doesn't seem like the right place for them.